modules/cart/src/Plugin/Block/CartBlock.php

```diff
@@ -206,11 +205,25 @@ protected function getCartViews(array $carts) {
 
   /**
    * {@inheritdoc}
-   *
-   * @todo Find proper cache tags to make this cacheable
    */
-  public function getCacheMaxAge() {
-    return 0;
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['cart']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
```

Comment:
A cache context's tags are only used when a cache context is "optimized away". E.g. when you have both `'user'` and `'user.permissions'` cache contexts, the `'user.permissions'` cache context is optimized away. Only then is `\Drupal\Core\Cache\Context\AccountPermissionsCacheContext::getCacheableMetadata()` used.

The `cart` cache context cannot ever be optimized away. So it's rather pointless for that cache context to even return cache tags.

See https://www.drupal.org/docs/8/api/cache-api/cache-contexts#optimizing.